From ec90b9e40bb1e50faf9f4ce59e58261e888d4f7e Mon Sep 17 00:00:00 2001
From: Sudip Mukherjee <sudipm.mukherjee@gmail.com>
Date: Sun, 11 Feb 2024 18:47:52 +0000
Subject: [PATCH] Support use of 7zip

Debian and Ubuntu are now using 7zip instead of p7zip. With 7zip the
archieve creation fails as it does not support '-l'.
This change will disable '-l' if anyone defines USE_7Z in the build flags.
Other builds who have not defined 'USE_7Z' are not affected.

Signed-off-by: Sudip Mukherjee <sudipm.mukherjee@gmail.com>

Origin: upstream, https://github.com/lxqt/lxqt-archiver/commit/ad850a5af053bbd5e6d4c73b0e699f4a76a1f9cb
Bug: https://github.com/lxqt/lxqt-archiver/issues/379
Bug-Ubuntu: https://launchpad.net/bugs/2052880
Last-Update: 2024-02-11
---
 CMakeLists.txt           | 4 ++++
 src/core/fr-command-7z.c | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0680105..50532d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,10 @@ project(lxqt-archiver)
 
 option(UPDATE_TRANSLATIONS "Update source translation translations/*.ts files" OFF)
 
+option(USE_7Z "Building with 7z" OFF)
+if(USE_7Z)
+    add_definitions(-DUSE_7Z)
+endif()
 
 set(GLIB_MINIMUM_VERSION "2.50.0")
 set(LIBFMQT_MINIMUM_VERSION "1.4.0")
diff --git a/src/core/fr-command-7z.c b/src/core/fr-command-7z.c
index 504f1e2..887fbcd 100644
--- a/src/core/fr-command-7z.c
+++ b/src/core/fr-command-7z.c
@@ -345,7 +345,10 @@ fr_command_7z_add (FrCommand     *comm,
 	if (spd_support) fr_process_add_arg (comm->process, "-spd");
 	fr_process_add_arg (comm->process, "-bd");
 	fr_process_add_arg (comm->process, "-y");
+#ifndef USE_7Z
+	/* Unlike p7zip, 7zip does not support '-l'. Define USE_7Z to disable '-l'. */
 	fr_process_add_arg (comm->process, "-l");
+#endif
 	add_password_arg (comm, comm->password, FALSE);
 	if ((comm->password != NULL)
 	    && (*comm->password != 0)
-- 
2.39.2

